tests/test.py

```diff
@@ -3823,6 +3842,155 @@ def testEmptyValues(self):
         self.assertEqual(row['binary'], b'')
 
 
+class TestOutdatedServiceConfigs(BaseTestCase, FailedRunProcessMixin):
+    def setUp(self):
+        return
+
+    def tearDown(self):
+        return
+
+    def testStartupWithoutVersionInConfig(self):
+        files = os.listdir('cmd/')
+        services = [i for i in files if os.path.isdir(os.path.join('cmd', i))]
+        self.assertTrue(services)
+
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            # generate configs for tests
+            subprocess.check_output(['configs/regenerate.sh', tmp_dir])
+
+            for service in services:
+                # remove first line that contains version
+                subprocess.check_output(['sed', '-i', '1d', os.path.join(tmp_dir, service + '.yaml')])
```

Comment:
Various UNIX systems have slightly different **sed** flavors with different command lines. For example, the built-in sed on macOS handles `-i` flag a bit differently. This may get annoying for developers who try running tests outside of Docker containers with Linux inside. (Though I don't know how common that is.)

I wonder if there is a better way to handle this. For example, we could teach the `configs/regenerate.sh` tool to generate configs without a version field (with `--no-version` option), or with a particular version value (`--version=0.100.500`). That way we could avoid unnecessary subprocess calls from tests.

Reply:
oh. first time I wanted to do it from python script but thought that it's will be too complicated with reading file into memory + replacing + dumping to file and decided to do it with sed. then thought that we can use 2-step replacing with sed: sed to temp file and rename but then remembered that he can do it with in place mode. chose second variant and forget about posix compatibility.

I don't want to extend `configs/regenerate.sh` because in general it will not be useful for users because they should use up to date config versions and script was created as helper to create all configs for current version of acra. Maybe will be good to add separate bash script for tests which will do this? Which will replace in 2-step way or forget about overhead and do it in python because config files has small size (biggest is 4.1k))...

Reply:
The version flags can be optional:

- no flags: generate configuration with current version
- `--no-version`: generate configuration without a version field
- `--version <V>`: generate configuration with specified version field

Though, in all cases we will generate a configuration _for_ the current version, just with different version field. Which is kinda hacky, I admit. So I don't know whether it's good to add these flags to regenerate.sh because we need them only for tests.

Anyway, this is a minor annoyance do it can be ignored if there is no one-liner fix for it.

I've seen (portable?) utilities like **yq** which can process YAML from command-line, but they are a bother to install. There should be Python modules which do the same, but that's another dependency to install.

Reply:
we already have pyYaml in dependencies for tests so I will rewrite it with replacing on tests side

support `--no-version`/`--version` in `configs/regenerate.sh` will meet the same problem: replace version with some tool like sed/etc and do it in a posix-compatible way and it will be useful only for tests.
add it on acra side will be complicated due to different handling flags by each binary or write in general part of code and it will be useful only for tests too ) And with such parameters we allow to use/generate configs without versions but it is not that we want. Because we added versions and force them to simplify discovering user's problems and misconfigurations in the future. And probably will drop support configs without versions in a future